server/replication/replication_mode.go

```diff
@@ -503,6 +503,10 @@ func (m *ModeManager) checkStoreStatus() [][]uint64 {
 		if s.IsRemoved() {
 			continue
 		}
+		// learner peers do not participate in major commit or vote, so it should not count in primary/dr as a normal store.
+		if s.GetRegionCount() == s.GetLearnerCount() {
```

Comment:
Is this comparison method confident? Do we need to check the actual status of each region?

Reply:
It's updated when the region state change, currently, I think it is believable. In addition, this does not require strict checks, because all states do not influence the data consistency, this state only let TiKV know how to decide the majority peers.